runtime/runtime.go

```diff
@@ -975,38 +1003,45 @@ func (r *interpreterRuntime) updateAccountCode(
 	if len(contractTypes) > 0 {
 		contractType := contractTypes[0]
 
-		contract, err := r.instantiateContract(
-			location,
-			contractType,
-			constructorArguments,
-			constructorArgumentTypes,
-			runtimeInterface,
-			runtimeStorage,
-			checker,
-			functions,
-			invocationRange,
-		)
+		if options.createContract {
 
-		if err != nil {
-			panic(err)
-		}
+			contract, err := r.instantiateContract(
+				location,
+				contractType,
+				constructorArguments,
+				constructorArgumentTypes,
+				runtimeInterface,
+				runtimeStorage,
+				checker,
+				functions,
+				invocationRange,
+			)
 
-		contractValue = interpreter.NewSomeValueOwningNonCopying(contract)
+			if err != nil {
+				panic(err)
+			}
+
+			contractValue = interpreter.NewSomeValueOwningNonCopying(contract)
+		}
 	}
 
-	address := common.Address(addressValue)
+	if options.createContract {
+		address := common.Address(addressValue)
 
-	contractValue.SetOwner(&address)
+		contractValue.SetOwner(&address)
+	}
 
 	// NOTE: only update account code if contract instantiation succeeded
 	wrapPanic(func() {
-		err = runtimeInterface.UpdateAccountCode(addressValue.ToAddress(), code, checkPermission)
+		err = runtimeInterface.UpdateAccountCode(addressValue.ToAddress(), code, options.checkPermission)
```

Comment:
As you can see here, the `checkPermission` parameter isn't used by the virtual machine anymore. Can we just remove it? 

https://github.com/dapperlabs/flow-go/blob/a323a57b52300345d617564d45a33c80574b973a/engine/execution/computation/virtualmachine/transactionContext.go#L291-L320

Reply:
Good idea, removed in 491e698c8e780e57c52c874cbbd86f3b23a34485